storm-core/src/jvm/storm/trident/topology/TridentTopologyBuilder.java

```diff
@@ -233,8 +234,9 @@ public StormTopology buildTopology() {
             }
             
             Map<String, Set<String>> batchToComponents = getBoltBatchToComponentSubscriptions(id);
-            for(String b: batchToComponents.keySet()) {
-                for(String comp: batchToComponents.get(b)) {
+            for(Map.Entry<String, Set<String>> entry: batchToComponents.entrySet()) {
+                String b = entry.getKey();
+                for(String comp: entry.getValue()) {
                     d.directGrouping(comp, TridentBoltExecutor.COORD_STREAM(b));
```

Comment:
Unless the variable is created to make the code easier to read, this could be simplified to 
d.directGrouping(comp, TridentBoltExecutor.COORD_STREAM(entry.getKey()));